apps/music/js/metadata/ogg.js

```diff
@@ -39,131 +43,194 @@ var OggMetadata = (function() {
+    readIdentificationHeader: function(blobview) {
+      var header = this.readPageHeader(blobview);
+      if (header.segment_table.length !== 1) {
+        throw new Error(
+          'ogg identification header expected as only packet of first page'
+        );
+      }
+      // Skip over the identification header.
+      blobview.advance(header.segment_table[0]);
+    },
+
+    /**
+     * Read the Vorbis Comment. Deal with multiple pages. Parse the
+     * picture block too.
+     *
+     * @param {BlobView} page The audio file being parsed.
+     * @return {Promise} A Promise that resolves with the completed metadata.
+     */
+    readVorbisComment: function(blobview) {
+      return new Promise(function(resolve, reject) {
+
+        this.done = function(metadata) {
+          LazyLoader.load('js/metadata/vorbis_picture.js').then(() => {
+            return VorbisPictureComment.parsePictureComment(metadata);
+          }).then(resolve);
+        }.bind(this);
+
+        this.readCommentHeader(blobview);
+      }.bind(this));
+    },
 
-    var sum = function(a, b) { return a + b; };
-    var comment_length = Array.reduce(header.segment_table, sum, 0);
+    /**
+     * Read the comment header of an Ogg container.
+     *
+     * @param {BlobView} bloview The audio file being parsed.
+     * @param {Metadata} metadata The (partially filled-in) metadata object.
+     */
+    readCommentHeader: function(blobview) {
+      var metadata = {};
+      var header = this.readPageHeader(blobview);
 
-    return new Promise((resolve, reject) => {
-      page.getMore(page.index, comment_length, (fullpage, err) => {
+      var sum = function(a, b) { return a + b; };
+      this.page_length = Array.reduce(header.segment_table, sum, 0);
+
+      blobview.getMore(blobview.index, this.page_length,
+                       function(fullpage, err) {
         if (err) {
-          reject(err);
-          return;
+          throw new Error(err);
         }
 
-        try {
-          // Look for a comment header from a supported codec
-          var first_byte = fullpage.readByte();
-          var valid = false;
-          switch (first_byte) {
-          case 3:
-            valid = fullpage.readBinaryText(6) === 'vorbis';
-            metadata.tag_format = 'vorbis';
-            break;
-          case 79:
-            valid = fullpage.readBinaryText(7) === 'pusTags';
-            metadata.tag_format = 'opus';
-            break;
+        // Look for a comment header from a supported codec
+        var first_byte = fullpage.readByte();
+        var valid = false;
+        switch (first_byte) {
+        case 3:
+          valid = fullpage.readBinaryText(6) === 'vorbis';
+          metadata.tag_format = 'vorbis';
+          break;
+        case 79:
+          valid = fullpage.readBinaryText(7) === 'pusTags';
+          metadata.tag_format = 'opus';
+          break;
+        }
+        if (!valid) {
+          throw new Error('malformed ogg comment packet');
```

Comment:
Since you're in the callback for `BlobView.getMore`, you can't throw here. You need to use `reject`, since exceptions can get lost here: https://github.com/mozilla-b2g/gaia/blob/master/shared/js/blobview.js#L67